src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql

```diff
@@ -1165,17 +1173,36 @@ AS
     AND     SeriesInstanceUid = ISNULL(@seriesInstanceUid, SeriesInstanceUid)
     AND     SopInstanceUid = ISNULL(@sopInstanceUid, SopInstanceUid)
 
-    IF (@@ROWCOUNT = 0)
-    BEGIN
-        THROW 50404, 'Instance not found', 1;
-    END
+    IF @@ROWCOUNT = 0
+        THROW 50404, 'Instance not found', 1    
 
     -- Deleting tag errors
+    DECLARE @deletedErrors AS TABLE
```

Comment:
Should we just declare this as a temporary table?
```sql
CREATE TABLE #deletedErrors (TagKey BIGINT);
```

Reply:
seems table variable has better performance than temp table. https://www.sqlshack.com/when-to-use-temporary-tables-vs-table-variables/. I'll change everything to table variable.